pkg/services/store/entity/key.go

```diff
@@ -14,35 +14,56 @@ type Key struct {
 }
 
 func ParseKey(key string) (*Key, error) {
-	// /<group>/<resource>/<namespace>(/<name>(/<subresource>))
-	parts := strings.SplitN(key, "/", 6)
-	if len(parts) < 4 {
-		return nil, fmt.Errorf("invalid key (expecting at least 3 parts): %s", key)
+
+	// /<group>/<resource>
+	parts := strings.SplitN(key, "/", 4)
```

Comment:
Not sure why we need to do this vs just splitting with a maximum of 6 up front and checking length of the returned slice?